src/index.ts

```diff
@@ -1,4 +1,78 @@
 import { Probot } from "probot";
+import fs from 'fs';
+import { V1Secret } from '@kubernetes/client-node/dist/gen/model/v1Secret';
+const k8s = require('@kubernetes/client-node');
+const { createAppAuth } = require("@octokit/auth-app");
+var cron = require('node-cron');
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+const k8sContext = kc.getCurrentContext();
+const k8snamespace = (() => {
+  if (k8sContext === 'inClusterContext') {
+    return fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/namespace', 'utf8');
+  } else {
+    return k8sContext.split('/')[0];
+  }
+})();
+console.info(`Operating in the namespace: ${k8snamespace}`);
+
+async function getToken(installId: Number) {
+
+  const appSecret = await k8sApi.readNamespacedSecret("peribolos-as-services", k8snamespace);
+  console.info(`Got the secret peribolos-as-service secrets`);
+  const appID = atob(appSecret.body.data.app_id);
+  const privateKey = atob(appSecret.body.data.private_key);
+  const auth = await createAppAuth({ appId: appID, privateKey: privateKey });
+  const appAuth = await auth({ type: "installation", installationId: installId });
+
+  return [appAuth.token, appAuth.expiresAt];
+};
+
+const patchSecret = async() => {
+
+  const resSecret = await k8sApi.listNamespacedSecret(k8snamespace,undefined,undefined,undefined,undefined,"app.kubernetes.io/created-by=peribolos")
+  for (var sec of resSecret.body.items){
```

Comment:
Should be rather a `.map` call, for efficiency. Array methods are always faster. https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/

You can also use `resSecret?.body?.items?.map` in case there's undefined value.

Reply:
Thanks for the info, i will make the changes